src/components/SiteCarousel.js

```diff
@@ -0,0 +1,151 @@
+// 3rd party libraries - core
+import React, {Component} from 'react';
+import {View, StyleSheet, ActivityIndicator, Dimensions, ImageBackground, TouchableOpacity} from 'react-native';
+import {Icon, Card, Text} from 'react-native-elements';
+import Swiper from 'react-native-swiper';
+import _ from 'lodash';
+
+import SitePreview from './SitePreview';
+
+// 3rd party libraries - additional
+
+// styles and language
+import {campsite} from '../locale.en';
+import {attemptToAddFavorite, attemptToRemoveFavorite, getSiteDetail} from "../actions";
+import {connect} from "react-redux";
+
+const {campsite_form: {accessibility_options}} = campsite;
+
+const SCREEN_WIDTH = Dimensions.get('window').width;
+
+
+class SiteCarousel extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {}
+    }
+
+    componentWillMount() {
+        const {displaySites, selectedSite} = this.props;
+
+        if(displaySites.length === 0){
+            return;
+        }
+
+        const currentSiteIndex = _.findIndex(displaySites, site => site.id === selectedSite.id);
+
+        this.setState({
+            initialIndex: currentSiteIndex > -1 ? currentSiteIndex : 0
+        });
```

Comment:
I don't like mixing React component state with our global Redux state, but this really seemed like a good use for it... we don't need to be firing off to the global store the specific index for the carousel. If we decide that we should, it's a simple refactor.

Reply:
Would that mean for other components that don’t require a state change from the store as well? Like toggles and such?